src/Broker/Broker.c

```diff
@@ -57,15 +69,25 @@ Broker_deliver_request(const enum RemoteInterface interface, const uint8_t* cons
     const Packetizer_PacketType packet_type = Packetizer_PacketType_Telemetry;
 #endif
 
-    const size_t packet_size = Packetizer_packetize(&packetizer_data,
+    const enum SystemBus bus_id = port_to_bus_map[interface];
+    enum PacketizerCfg packetizer_name = bus_to_packetizer_cfg[bus_id];
+
+    unsigned header_size = SPACE_PACKET_PRIMARY_HEADER_SIZE;
+    if(packetizer_name == PACKETIZER_THIN) {
+        header_size = THIN_PACKET_PRIMARY_HEADER_SIZE;
+    }
```

Comment:
What's better, the above approach? Or should the header size be stored in the 'packetizers' array?

Reply:
Yes, this should be stored in an array. I would prefer to avoid any packetizer-specific IFs or defines, if possible.